tests/main/security-device-cgroups/task.yaml

```diff
@@ -77,50 +80,62 @@ execute: |
 
-    echo "And other devices are not shown as assigned to the snap"
-    udevadm info "$OTHER_UDEVADM_PATH" | not MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
+        echo "And other devices are not shown as assigned to the snap"
+        udevadm info "$OTHER_UDEVADM_PATH" | not MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
+        udevadm info "$OTHER_UDEVADM_PATH" | not MATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
 
-    echo "When a snap command is called"
-    test-snapd-sh.sh -c 'true'
+        echo "When a snap command is called"
+        test-snapd-sh.sh -c 'true'
 
-    echo "Then the device is shown in the snap device list"
-    MATCH "$DEVICE_ID" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        echo "Then the device is shown in the snap device list"
+        MATCH "$DEVICE_ID" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
 
-    echo "And other devices are not shown in the snap device list"
-    not MATCH "$OTHER_DEVICE_ID" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        echo "And other devices are not shown in the snap device list"
+        not MATCH "$OTHER_DEVICE_ID" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
 
-    echo "But existing nvidia devices are in the snap's device cgroup"
-    MATCH "c 195:0 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
-    MATCH "c 195:255 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
-    MATCH "c 247:0 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        echo "But existing nvidia devices are in the snap's device cgroup"
+        MATCH "c 195:0 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        MATCH "c 195:255 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        MATCH "c 247:0 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
 
-    echo "But nonexisting nvidia devices are not"
-    not MATCH "c 195:254 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        echo "But nonexisting nvidia devices are not"
+        not MATCH "c 195:254 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
 
-    echo "But the existing uhid device is in the snap's device cgroup"
-    MATCH "c 10:239 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
+        echo "But the existing uhid device is in the snap's device cgroup"
+        MATCH "c 10:239 rwm" < /sys/fs/cgroup/devices/snap.test-snapd-sh.sh/devices.list
 
-    # TODO: check device unassociated after removing the udev file and rebooting
+        echo "Once the snap is removed, the current tags are automatically removed"
+        snap remove test-snapd-sh
+        udevadm info "$UDEVADM_PATH" | not MATCH "E: CURRENT_TAGS=.*snap_test-snapd-sh_sh"
+
+        rm -f /etc/udev/rules.d/70-snap.test-snapd-sh.rules
+        REBOOT
+    fi
+
+    udevadm info "$UDEVADM_PATH" | not MATCH "E: TAGS=.*snap_test-snapd-sh_sh"
```

Comment:
can we use `NOMATCH` here?

Reply:
sorry, because copy&paste. Updated all the "not MATCH" in that test.
Thanks for catching this